### ISSUE TITLE
support API for open and run a notebook file

### ISSUE THREAD

**Issue by Roger-luo** [fonsp/Pluto.jl#435]
currently we need do quite a few lines to mutate a `Session`, then run `Pluto.run(s)` to open a notebook directly.

This has some inconvenience:

1. it's a bit cumbersome to open just one notebook
2. can't launch browser at the notebook url

I tried to polish the `Pluto.run` function so I can split each step to spawn the server process into separate functions, but I eventually realize it's too complicate for me to do it... so maybe @fonsp would know how to do it properly.

basically I'm thinking about launching the notebook url in browser when user calling `pluto open notebook.jl` with CLI instead of the server url. Also it would be nice to have an interface that let user open a notebook directly in REPL.

**Comment by fonsp** [comment]
How about we split out the code to generate the `/edit?id=...` URL into a function?

I feel like having this option to open notebooks directly in the CLI is already a solution - we don't need it in Pluto.run.

**Comment by Roger-luo** [comment]
> How about we split out the code to generate the /edit?id=... URL into a function?

this sounds good. I guess I can then just call it from CLI and hook the rest part myself.

> I feel like having this option to open notebooks directly in the CLI is already a solution - we don't need it in Pluto.run.

I think I'm more suggesting to have `Pluto.open` actually support it, given it's already an API that is a bit lower level than `Pluto.run` currently. I think in principle,
we can just have `Pluto.open(notebook; kwargs...)` as a new interface, which shares some common code with `Pluto.run` internally.

**Comment by fonsp** [comment]
Yes that does make sense :) Feel free to submit a draft

**Comment by fonsp** [comment]
#601
